misc/ballerinalang-data-mapper/src/main/java/org/ballerinalang/datamapper/AIDataMapperCodeActionUtil.java

```diff
@@ -130,8 +165,37 @@
             Range newTextRange = CommonUtil.toRange(diagnostic.location().lineRange());
             LinePosition linePosition = diagnostic.location().lineRange().startLine();
 
+            SyntaxTree syntaxTree = context.workspace().syntaxTree(context.filePath()).get();
+
             // get the symbol at cursor type
-            Symbol symbolAtCursor = semanticModel.symbol(srcFile.get(), linePosition).get();
+            Symbol symbolAtCursor;
+            Optional<Symbol> symbolFromModel = semanticModel.symbol(srcFile.get(), linePosition);
+            if (symbolFromModel.isPresent()) {
+                symbolAtCursor = symbolFromModel.get();
+                lftTypeSymbol = (Symbol) props.get(LEFT_SYMBOL_INDEX).value();
+                rhsTypeSymbol = (Symbol) props.get(RIGHT_SYMBOL_INDEX).value();
+
+                foundTypeLeft = lftTypeSymbol.getName().orElse("");
+                foundTypeRight = rhsTypeSymbol.getName().orElse("");
+            } else {
+                // to get the symbol when type casting is done
+                Optional<Node> nodeAtCursor = findExpressionInTypeCastNode(newTextRange, syntaxTree);
+                if (nodeAtCursor.isPresent()) {
+                    symbolFromModel = semanticModel.symbol(nodeAtCursor.get());
+                    if (symbolFromModel.isPresent()) {
+                        symbolAtCursor = symbolFromModel.get();
+                        lftTypeSymbol = (Symbol) props.get(RIGHT_SYMBOL_INDEX).value();
+                        rhsTypeSymbol = (Symbol) props.get(LEFT_SYMBOL_INDEX).value();
+
+                        foundTypeLeft = lftTypeSymbol.getName().orElse("");
+                        foundTypeRight = rhsTypeSymbol.getName().orElse("");
+                    } else {
+                        return fEdits;
+                    }
+                } else {
+                    return fEdits;
+                }
```

Comment:
You can use the early return and reduce the nested if

Reply:
modified the code according to the comment